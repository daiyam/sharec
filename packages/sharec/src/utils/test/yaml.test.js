const path = require('path')
const { readFileSync } = require('fs')
const { fromYaml, toYaml } = require('../yaml')

const jsonFixture = require('fixtures/eslint/json/eslintrc_01.json')
const yamlFixture = readFileSync(
  path.resolve(__dirname, '../../../test/fixtures/eslint/yaml/eslintrc_01.yml'),
  'utf8',
)

describe('utils > yaml', () => {
  describe('fromYaml', () => {
    it('should convert yaml string to json', () => {
      expect(fromYaml(yamlFixture)).toMatchSnapshot()
    })
  })

  describe('toYaml', () => {
    it('should convert json or js-object to yaml string', () => {
      expect(toYaml(jsonFixture)).toMatchSnapshot()
    })
  })
})
