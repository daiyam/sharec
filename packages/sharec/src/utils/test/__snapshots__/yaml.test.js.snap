// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils > yaml fromYaml should convert yaml string to json 1`] = `
Object {
  "env": Object {
    "browser": true,
    "es6": true,
    "jest": true,
    "node": true,
  },
  "extends": "standard",
  "globals": Object {
    "bar": true,
    "foo": true,
  },
  "parserOptions": Object {
    "ecmaFeatures": Object {
      "spread": true,
    },
    "ecmaVersion": 8,
    "parser": "babel-eslint",
    "sourceType": "module",
  },
  "rules": Object {
    "max-len": Array [
      "warn",
      80,
      2,
      Object {
        "ignoreComments": false,
        "ignoreStrings": true,
        "ignoreTemplateLiterals": false,
        "ignoreUrls": true,
      },
    ],
  },
}
`;

exports[`utils > yaml toYaml should convert json or js-object to yaml string 1`] = `
"env:
  browser: true
  es6: true
  node: true
  jest: true
globals:
  foo: true
  bar: true
extends: standard
rules:
  max-len:
    - warn
    - 80
    - 2
    - ignoreUrls: true
      ignoreComments: false
      ignoreStrings: true
      ignoreTemplateLiterals: false
parserOptions:
  parser: babel-eslint
  ecmaVersion: 8
  ecmaFeatures:
    spread: true
  sourceType: module
"
`;
